firstvoices/backend/search/utils/object_utils.py

```diff
@@ -108,63 +108,70 @@ def hydrate_objects(search_results, request):
-                    "entry": StorySerializer(
-                        story,
-                        context={
-                            "request": request,
-                            "view": "search",
-                            "site": story.site,
-                        },
-                    ).data,
-                }
-            )
+        try:
+            # Handling DictionaryEntry objects
+            if ELASTICSEARCH_DICTIONARY_ENTRY_INDEX in obj["_index"]:
+                dictionary_entry = get_object_by_id(
+                    dictionary_objects, obj["_source"]["document_id"]
+                )
+
+                # Serializing and adding the object to complete_objects
+                complete_objects.append(
+                    {
+                        "score": obj["_score"],
+                        "type": dictionary_entry.type.lower(),  # 'word' or 'phrase' instead of 'dictionary_entry'
+                        "entry": DictionaryEntryDetailSerializer(
+                            dictionary_entry,
+                            context={
+                                "request": request,
+                                "view": "search",
+                                "site": dictionary_entry.site,
+                            },
+                        ).data,
+                    }
+                )
+            elif ELASTICSEARCH_SONG_INDEX in obj["_index"]:
+                song = get_object_by_id(song_objects, obj["_source"]["document_id"])
+
+                # Serializing and adding the object to complete_objects
+                complete_objects.append(
+                    {
+                        "score": obj["_score"],
+                        "type": "song",
+                        "entry": SongSerializer(
+                            song,
+                            context={
+                                "request": request,
+                                "view": "search",
+                                "site": song.site,
+                            },
+                        ).data,
+                    }
+                )
+            elif ELASTICSEARCH_STORY_INDEX in obj["_index"]:
+                story = get_object_by_id(story_objects, obj["_source"]["document_id"])
+
+                # Serializing and adding the object to complete_objects
+                complete_objects.append(
+                    {
+                        "score": obj["_score"],
+                        "type": "story",
+                        "entry": StorySerializer(
+                            story,
+                            context={
+                                "request": request,
+                                "view": "search",
+                                "site": story.site,
+                            },
+                        ).data,
+                    }
+                )
+        except AttributeError as e:
+            if "has no site" in str(e):
```

Comment:
Handling a very specific case, so if there are any new errors pop up, we can track them.

Reply:
That is a great idea to prevent the app from crashing from any unknown bugs. I will add that. Thank you